src/chatgpt-api.ts

```diff
@@ -67,6 +75,11 @@ export class ChatGPTAPI {
 
     this._accessTokenCache = new ExpiryMap<string, string>(accessTokenTTL)
 
+    if (proxyUrl) {
+      this._proxyAgent = new ProxyAgent(proxyUrl)
+      setGlobalDispatcher(this._proxyAgent)
```

Comment:
I don't think we want to set a global dispatcher which will affect the whole process; rather, the proxy should only affect our usage of `fetch` within this module.